src/compiler/CombineJavascriptsCompiler.ts

```diff
@@ -0,0 +1,160 @@
+        private readonly dependencyPaths: string[],
+        private readonly assetsPath: string,
+        private readonly mainRepoDir: string,
+        private readonly isProduction: boolean
+    ) {
+        this.pathToEntryFile = path.join(
+            this.assetsPath,
+            CombineJavascriptsCompiler.ENTRY_FILE_NAME
+        );
+    }
+
+    compile(): Promise<CompilationResult> {
+        return new Promise<CompilationResult>((resolve, reject) => {
+            const generatedDir = CombineJavascriptsCompiler.getOutputDir(this.assetsPath);
+            const outputFile = path.join(
+                generatedDir,
+                CombineJavascriptsCompiler.OUTPUT_FILE_NAME
+            );
+
+            if (!fs.existsSync(this.pathToEntryFile)) {
+                this.cleanOutput(outputFile);
+                return resolve(CompilationResult.CHANGED);
+            }
+
+            if (!fs.existsSync(generatedDir)) {
+                fs.mkdirSync(generatedDir);
+            }
+
+            const start = new Date().getTime();
+            console.log(`⟲ [${this.pluginName}] starting to combine javascripts...`);
+
+            this.combineJavascripts().then(() => {
+                let end = new Date().getTime();
+                console.log(
+                    GREEN_CHECK,
+                    `[${
+                        this.pluginName
+                    }] Combining javascripts finished (${formatDuration(end - start)})`
+                );
+                return resolve(CompilationResult.CHANGED);
+            })
+            .catch((err) => {
+                console.error(cerr`${err}`);
+                reject(`[${this.pluginName}] Failed to combine javascripts`);
+            });
+        })
+        .catch((err) => {
+            throw Error(err);
+        });
+    }
+
+    public static getOutputDir(assetsPath: string): string {
+        return path.resolve(assetsPath, CombineJavascriptsCompiler.OUTPUT_DIR);
+    }
+
+    private combineJavascripts() {
+        return new Promise((resolve, reject) => {
+            // @ts-ignore
+            webpack(this.getCombineJavascriptWebpackConfig(), (err, stats) => {
+                if (err) {
+                    reject(err);
+                } else if (stats.hasErrors()) {
+                    reject(stats.toString());
+                } else {
+                    resolve();
+                }
+            });
+        });
+    }
+
+    private getCombineJavascriptWebpackConfig(): Configuration {
+        const config: Configuration = {
+            context: path.resolve(
+                this.assetsPath
+            ),
+            externals: {
+                jquery: 'jQuery'
+            },
+            resolveLoader: {
+                modules: [path.resolve(__dirname, '../../', 'node_modules')],
+            },
+            entry: {
+                compressed: this.getCombineJavascriptSources()
+            },
+            module: {
```

Comment:
Probably we should add some optimizations as well too keep the size of the file small. Something like [this](https://github.com/collaborationFactory/cplace-asc/blob/improvement/PFM-ISSUE-9113-move-combineJavascripts-task-from-build-plugin-to-cplace-asc/src/compiler/VendorCompiler.ts#L247)